app/src/pages/api/attachment/upload-url.ts

```diff
@@ -0,0 +1,29 @@
+import S3 from 'aws-sdk/clients/s3'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  console.log("Using BUCKET_NAME:", process.env.BUCKET_NAME);
+  const s3 = new S3({
+    apiVersion: '2006-03-01',
+    accessKeyId: process.env.CL_AWS_ACCESS_KEY,
+    secretAccessKey: process.env.CL_AWS_SECRET_ACCESS_KEY,
+    region: process.env.AWS_REGION
+  })
+
+  const post = await s3.createPresignedPost({
+    Bucket: process.env.BUCKET_NAME,
+    Fields: {
+      key: req.query.file,
+      'Content-Type': req.query.fileType,
+    },
+    Expires: 60, // seconds
+    Conditions: [
+      ['content-length-range', 0, 1048576], // up to 1 MB
```

Comment:
The `1048576` constant and `up to 1 MB` comment could be unnecessary if we set up :)

For example, we can create a constant for megabyte:

```
const MEGABYTE = 10 ** 6;
```

Then, our condition will be clearer by itself:

```
Conditions: [
    ['content-length-range', 0, 1 * MEGABYTE]
]
```

And we can always multiply the megabyt if we need to increase the size :)